src/hotspot/share/prims/jvm.cpp

```diff
@@ -1269,87 +1269,6 @@ JVM_ENTRY(jobject, JVM_GetProtectionDomain(JNIEnv *env, jclass cls))
 JVM_END
 
 
-// Returns the inherited_access_control_context field of the running thread.
```

Comment:
There's some code in this file in 
static void trace_class_resolution_impl(Klass* to_class, TRAPS) {

That does this:

```
    while (!vfst.at_end()) {
      Method* m = vfst.method();
      if (!vfst.method()->method_holder()->is_subclass_of(vmClasses::ClassLoader_klass())&&
          !vfst.method()->method_holder()->is_subclass_of(access_controller_klass) &&
          !vfst.method()->method_holder()->is_subclass_of(privileged_action_klass)) {
        break;
      }
      last_caller = m;
      vfst.next();
    }
```

Is this dead code that should be removed?

Reply:
This tracing skips ClassLoader frames, you'll continue to see these when using Class.forName.

Reply:
but you won't see access_controller_klass or priviledged_action_klass frames, so no need to skip them?

Reply:
Right, although you might have to wait until there is more cleanup in the JDK code before they disappear completely. To clarify, most uses of privileged actions are only done when a SecurityManager is set but there some cases where they execute unconditionally. In the Class/ClassLoader then I think they are done conditionally so you should be okay. Are there tests for `-Xlog:class+resolve=debug` that would fail if we had a bug in this tracing code?

Reply:
So you're saying there will still be these frames in the stack that we want to skip?  Okay, we'll clean all that out later then.

Reply:
I have the changes to Class/ClassLoader to propose as soon as this JEP integrates. If that can go in quickly then it would remove any concerns in advance of your cleanup to remove the filtering of these frames (although I don't think there is an issue as this code path is conditional based on whether a SM is set).